executor/modes/local/runner.py

```
@@ -0,0 +1,407 @@
+        self._shutdown_event = asyncio.Event()
+
+        # Setup signal handlers
+        self._setup_signal_handlers()
+
+    def _setup_signal_handlers(self) -> None:
+        """Setup signal handlers for graceful shutdown."""
+        # Note: Signal handlers in asyncio should be set in the main thread
+        # We'll handle this in start() method
+        pass
+
+    def _handle_signal(self, signum: int, frame: Any) -> None:
```

Ruff (0.14.14)

Unused method argument: `frame`

(ARG002)